### PR TITLE
pkg/{cfgstruct,identity}: replace seperator in default values when path tag set

### PR DESCRIPTION
**Pull request by bryanchriswhite**
What: Add `path:"true"` flag to identity config fields which represent file paths and replace `/` with `os.Separator` in defaults where `path:"true"` tag is set.

Why: [V3-2979](https://storjlabs.atlassian.net/browse/V3-2979?atlOrigin=eyJpIjoiMzQ0OWExNDBjNWJkNDZhY2FjZmMxMTk0NWRlNjgyNzEiLCJwIjoiaiJ9)

Please describe the tests: N/A
 
Please describe the performance impact: N/A

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/cfgstruct/bind.go [hunk 1]
```diff
@@ -260,6 +260,10 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value, vars map[string
 				check(err)
 				flags.Float64Var(fieldaddr.(*float64), flagname, val, help)
 			case reflect.TypeOf(string("")):
+				if field.Tag.Get("path") == "true" {
+					// NB: conventionally unix path separators are used in default values
+					def = strings.ReplaceAll(def, "/", string(filepath.Separator))
```
Comment (egonelbre) [inline pkg/cfgstruct/bind.go]: There are `filepath.FromSlash` and `filepath.ToSlash`

Comment (bryanchriswhite) [reply pkg/cfgstruct/bind.go]: go to know thanks!

**Comment by bryanchriswhite** [comment]
Should we also use `path:"true" for config fields which use `$CONFDIR` and/or other filepath fields?

**Comment by ifraixedes** [comment]
@bryanchriswhite 
> Should we also use path:"true" for config fields which use $CONFDIR and/or other filepath fields?

I would say yes.

**Pull request merged**